CMakeLists.txt

```diff
@@ -49,13 +49,14 @@ zephyr_library_named(zephyr)
 
 zephyr_include_directories(
   kernel/include
-  arch/${ARCH}/include
+  ${ARCH_DIR}/${ARCH}/include
   ${SOC_DIR}/${ARCH}/${SOC_PATH}
   ${SOC_DIR}/${ARCH}/${SOC_PATH}/include
   ${SOC_DIR}/${ARCH}/${SOC_FAMILY}/include
   ${BOARD_DIR}
   include
   include/drivers
+  ${ARCH_INCL_DIR}
```

Comment:
I'm not seeing the need for ARCH_INCL_DIR.

Would it be possible to run

zephyr_include_directories(..)

from within the out-of-tree ARCH ?

Reply:
You are right that the same can be achieved by adding the following line:
`zephyr_include_directories(../../include)` to the out-of-tree file {ARCH_ROOT}/arch/{ARCH}/CMakeLists.txt

The trade-off beeing that this then requires arch's that are moved out-of-tree to add that line. With the ${ARCH_INCL_DIR} it does not. Both solutions will work.

I will update the PR with your suggestions